python/doc/source/other_systems.rst

```diff
@@ -0,0 +1,104 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. currentmodule:: pyarrow
+.. _other_systems:
+
+Using Arrow with other systems
+=============================================
+
+Timestamps
+----------
+
+Timestamps are data structures that mark a particular point in time
+and we want to be able to order them, regardless of where
+they originated.
+For human consumption timestamps are usually specified by the date
+together with the time of day, often using the local time zone.
+The problem with this scheme is that if things need to be ordered
+by time across multiple time zones, using local time can be ambiguous.
+As many areas in the Western world still use Daylight Saving Time it
+can be impossible to convert local time to UTC during the period in
+the fall (in the Northern Hemisphere) when local time is moved back.
+Therefore timestamp values should always be collected, stored and
+communicated in UTC to avoid this ambiguity.
+Optionally the time zone of the measurement can be stored and 
+communicated with it, but often the the current
+time zone of the user can be used to present a time value in local
+time.   
```

Comment:
While not untrue, the DST issues and choice to represent data localized to UTC are a choice to be made, and we made the choice in Arrow to require UTC localization if data is time zone aware. 

The problem vs. Spark semantics is the interpretation of data that has no time zone / is not time zone aware. Nowhere in this document is the concept of session / system time zone mentioned, nor are the difference between time zone naive and time zone aware timestamps, which are the core issues at play in ARROW-1425, and were discussed on the mailing list (thread: https://lists.apache.org/thread.html/d946a0df1bd63c05b002b0feb0808eee7741e3337bc0484a2d5efa80@%3Cdev.arrow.apache.org%3E)